.github/workflows/node-build.yml

```
@@ -0,0 +1,316 @@
+name: Node Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - main
+      - release/v*
+
+jobs:
+  prerequisite:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ./.github/workflows/rafiki/env-setup
+    - run: pnpm checks
+
+  backend:
+    runs-on: ubuntu-latest
+    needs: prerequisite
+    timeout-minutes: 25
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/workflows/rafiki/env-setup
+      - run: pnpm --filter backend build:deps
+      - run: NODE_OPTIONS=--max-old-space-size=4096 pnpm --filter backend test:ci
+      - name: AsyncAPI extension
+        run: |
+          echo "{\"extends\":[\"spectral:oas\",\"spectral:asyncapi\"]}" >> .spectral.json
+      - name: Validate Open API specs
+        run: |
+          npx @stoplight/spectral-cli lint ./packages/backend/src/openapi/*.yaml
```

```suggestion
          npx @stoplight/spectral-cli lint ./packages/backend/src/openapi/specs/*.yaml
```
my b for requesting change again, I just merged a bit of an overhaul w the openapi specs https://github.com/interledger/rafiki/pull/2530